### PR TITLE
PA-1491: Skip log collection in case of OCP

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
- [x] Skip log collection in case of OCP

**Which issue(s) this PR fixes** (optional)
Closes #PA-1491

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kshithijiyer-px** [comment]
Pass run: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/upgrade-tests/job/Px-Backup-with-Stork-Upgrade/job/Px-Backup+Stork-Upgrade-ocp-px/15/console

**Review by mkoppal-px — APPROVED** [review]
LGTM

File: tests/common.go [hunk 1]
```diff
@@ -6161,7 +6161,14 @@ func CollectLogsFromPods(testCaseName string, podLabel map[string]string, namesp
 
 	// Check to handle cloud based deployment with 0 master nodes
 	if len(node.GetMasterNodes()) == 0 {
-		log.Warnf("Skipping pod log collection for pods with [%s] label in test case [%s]", logLabel, testCaseName)
+		log.Warnf("Skipping pod log collection for pods with [%s] label in test case [%s] as it's cloud cluster", logLabel, testCaseName)
+		return
+	}
+
+	// In case of ocp if ssh_key is given skip log collection
+	sshKey := os.Getenv("TORPEDO_SSH_KEY")
```
Comment (kphalgun-px) [inline tests/common.go:6169]: Does the Torpedo pod get deployed on the master node in the case of OCP?

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/common.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/common.go:6169]: Yes

Comment (lsrinivas-pure) [inline tests/common.go:6169]: even for normal vsphere runs many PX jobs pass TORPEDO_SSH_KEY.

Comment (kshithijiyer-px) [reply tests/common.go:6169]: What we observed was that our runs were failing on ocp due to it not being able to ssh to collect log pods. This blocks the upgrade jobs from running. Currently from the backup end we don't have a job which has this parameter other than the ocp job.

Comment (lsrinivas-pure) [reply tests/common.go:6169]: generally we pass nodedriver=openshift for ocp jobs. can you use that variable to check if it is ocp or not

Comment (kshithijiyer-px) [reply tests/common.go:6169]: I checked this job which you share the problem is I don't see `schedular=openshift` being used anywhere in the code
https://jenkins.pwx.dev.purestorage.com/job/3.1.0-dev-jobs/job/ocp-dev-2/build?delay=0sec

Comment (kshithijiyer-px) [reply tests/common.go:6169]: Got a one which @lsrinivas-pure  is using the flag https://jenkins.pwx.dev.purestorage.com/job/Torpedo/view/vsphere-dmthin/job/OCP-PoolExpand-1/36/consoleFull. Changing the code.

Comment (kshithijiyer-px) [reply tests/common.go]: Done

File: tests/common.go [hunk 2]
```diff
@@ -6161,7 +6162,13 @@ func CollectLogsFromPods(testCaseName string, podLabel map[string]string, namesp
 
 	// Check to handle cloud based deployment with 0 master nodes
 	if len(node.GetMasterNodes()) == 0 {
-		log.Warnf("Skipping pod log collection for pods with [%s] label in test case [%s]", logLabel, testCaseName)
+		log.Warnf("Skipping pod log collection for pods with [%s] label in test case [%s] as it's cloud cluster", logLabel, testCaseName)
+		return
+	}
+
+	// In case of ocp skip log collection
+	if Inst().S.String() != openshift.SchedName {
```
Comment (kphalgun-px) [inline tests/common.go:6170]: == ?

Comment (lsrinivas-pure) [reply tests/common.go:6170]: I think it should be ==

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/common.go (hunk 2 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Thanks for pointing out saw that was fixing it.

**Comment by kshithijiyer-px** [comment]
Yes was changing it only.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by kshithijiyer-px** [comment]
Successful run with @lsrinivas-pure's changes https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/upgrade-tests/job/Px-Backup-with-Stork-Upgrade/job/Px-Backup+Stork-Upgrade-ocp-px/18/

**Review by ak-px — APPROVED** [review]
LGTM

**Pull request merged**